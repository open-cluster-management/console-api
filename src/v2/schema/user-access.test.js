/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 * Copyright (c) 2020 Red Hat, Inc.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('User Access Resolver', () => {
  test('Correctly Resolves User Access Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
          {
            userAccess(resource:"pods", action:"delete", namespace:"", apiGroup:"")
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));
});
