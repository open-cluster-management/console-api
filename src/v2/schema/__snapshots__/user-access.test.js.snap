// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Access Resolver Correctly Resolves User Access Query 1`] = `
Object {
  "data": Object {
    "userAccess": Object {
      "userAccess": Object {
        "allowed": true,
        "reason": "RBAC: allowed by ClusterRoleBinding \\"oidc-admin-binding\\" of ClusterRole \\"cluster-admin\\" to User \\"https://127.0.0.1:9443/oidc/endpoint/OP#admin\\"",
      },
    },
  },
}
`;
